### PR TITLE
Movable and resizable panels

### LINKED ISSUE #1222: Improve tilemap construction menu
## Summary

the current construction menu has the following issues : 
- can't be moved around
- can't be resized
- display partially some of the items on the right column

See the following picture 
image

## Goal 

implement the following features : 
- move around menu
- resize it, when resizing, the grid of items should adapt too, filling the empty place with new columns
- by default, make it big enough to display correctly both column.

### PR DESCRIPTION
**Pull request by iamteapot422**
# Summary
Imports yasirkula's [dynamic panels](https://github.com/yasirkula/UnityDynamicPanels). It's features include:

- Resizable and movable panels with flexible settings
- Docking
- Merging several panels into one
- Icons in names for panels
- API

Dynamic panel prefab is modified to take less space in the header part

https://github.com/RE-SS3D/SS3D/assets/77565416/8efba392-3226-491d-90e0-0fc46afdb586

# Related issues/PRs 
Closes #1222

### CODE REVIEW

**Comment by iamteapot422** [comment]
Grid's constraint count now changes to fit grid's width




https://github.com/RE-SS3D/SS3D/assets/77565416/fdd2ed31-a1f5-4f49-8bbc-4184246d1ea6

**Comment by stilnat** [comment]
image

Looks cool but I only ever get an empty windows instead of construction.

**Review by stilnat — DISMISSED** [review]
Looks good to me, waiting a bit for John's change before merge

File: Assets/Scripts/SS3D/Systems/Tile/TileMapCreator/AssetGridManager.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+
+/// <summary>
+/// Component, that changes GridLayoutGroup contstraintCount to fit it's width
+/// </summary>
+[RequireComponent(typeof(GridLayoutGroup))]
+public class AssetGridManager : MonoBehaviour
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Systems/Tile/TileMapCreator/AssetGridManager.cs]: put it into an appropriate namespace and document it a bit more, maybe with an example. You can describe how it act on the construction menu.

**Review by stilnat — CHANGES_REQUESTED** [review]
small change and doc. Also fix conflicts.

**Comment by cosmiccoincidence** [comment]
Also, merge in master and resolve file conflicts please.

**Review by stilnat — APPROVED** [review]
good work

**Pull request merged**